src/main/java/com/three_tech_solutions/slot_app/services/implementations/StudentServiceImpl.java

```diff
@@ -73,4 +76,19 @@ private boolean planTypeIsSpecificDay(CreateStudentRequest studentDTO) {
     private boolean planTypeIsBeginningOfMonth(CreateStudentRequest studentDTO) {
         return studentDTO.getPlanType().equals(PlanType.PRINCIPIO_DE_MES);
     }
+    public void deleteStudent (UUID studentId){
+
+        Optional<Student> studentOptional = studentRepository.findById(studentId);
+
+        if (studentOptional.isPresent()) {
+            Student student = studentOptional.get();
+            if (!student.isEnabled()) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "El estudiante ya está eliminado.");
+            }
+            student.setEnabled(false);
+            studentRepository.save(student);
+        } else {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "El estudiante no existe.");
+        }
+    }
```

Comment:
Esto lo podriamos refactorizar utilizando las funciones lambdas y aprovechando los metodos de la API Optional


```suggestion
    public void deleteStudent(UUID studentId) {
        studentRepository.findById(studentId)
                .map(student -> {
                    if (!student.isEnabled()) {
                        throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "El estudiante ya está eliminado.");
                    }
                    student.setEnabled(false);
                    return studentRepository.save(student);
                })
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "El estudiante no existe."));
    }
```

De todas formas, lo tuyo funciona y estuviste muy bien en crear una validacion por si el estudiante ya estaba eliminado 🥹👍